cartoframes/client/sql_client.py

```diff
@@ -167,6 +167,9 @@ def drop_table(self, table_name):
         query = 'DROP TABLE IF EXISTS {0};'.format(table_name)
         return self.execute(query)
 
+    def _create_context(self):
+        return context.create_context(self._creds)
```

Comment:
What about the session?

Reply:
Session is inside of Credentials class

Reply:
OK, but check this with @andy-esch, because I'm not sure if it makes 100% sense

Reply:
Yeah, he is added as reviewer. 

We defined that in our doc. I think nothing changes: before we pass `session` to the `carto-python` clients thought `auth.Context` class. As we are going to remove it, now we will pass it thought Credentials class

Reply:
Aha. I'm thinking about what happens with the session when you load your creds from an external file, maybe we could pass the `session` in the `from_file` method.

Reply:
I thought about that: the session object persists data from previous requests using `python-requests`. I am not really sure about that, and as we are not saving it before, I decided not to add it to the `save` method. Maybe @alrocar & @andy-esch can give us more info

Reply:
The session object was originally supported to allow non-secure HTTPS requests ([see](https://github.com/CartoDB/carto-python/issues/59#issuecomment-316439017) this comment). This is a common case in on-premises installations, were we force https connections but the client might not have a valid SSL cert but a self-signed one.

Maybe we could just change the whole `session` object by just a param `verify_ssl_cert=True` that internally would create a session if is set to False.

That way, we'll just persist the `verify_ssl_cert` instead of the whole session, which seems an overkill to me. Also having just that param makes the `concept` easier to understand, the other way looks more like a hack.

Reply:
For use cases on `session`, I haven't seen any demand outside of verify ssl. I feel cautious about changing it too much though (like offering the flag alrocar is proposing) because there may be other `session` uses we don't know about..

As far as persisting the `session`s object, I haven't heard of any demand for doing that, so I say let's not save it until we get a good feature request for that.

Reply:
> As far as persisting the sessions object, I haven't heard of any demand for doing that, so I say let's not save it until we get a good feature request for that.

:+1:  I agree. If we don't have any real reason to change, I will keep it as it is

Reply:
At the same time... If it would save people time to store configs like that, that would be great.

Reply:
And what happens with the `from_file` method and a custom session?

Reply:
Right now it would be:
```
credentials = Credentials.from_file()
credentials.session = session
```

Reply:
Why not:

```py
credentials = Credentials.from_file(session=session)
```

?